src/Reference/HasOne.php

```diff
@@ -228,18 +236,14 @@ public function ref($defaults = []) : Model
                 $m->addHook('afterSave', function ($m) {
                     $this->owner[$this->our_field] = $m[$this->their_field];
                 });
-        } else {
-            if ($this->owner[$this->our_field]) {
-                $m->tryLoad($this->owner[$this->our_field]);
-            }
-
-            return
-                $m->addHook('afterSave', function ($m) {
-                    $this->owner[$this->our_field] = $m->id;
-                });
+        }
+        if ($this->owner[$this->our_field]) {
+            $m->tryLoad($this->owner[$this->our_field]);
         }
 
-        // can not load referenced model or set conditions on it, so we just return it
-        return $m;
+        return
```

Comment:
not completely sure about this change here. what if $this->our_field is not defined at all? then this will fail.
I understand why you split old if/else block into two if blocks, but this change here changes more than that.

Reply:
looking at the previous version, code will never arrive here : https://github.com/atk4/data/blob/0036a94d140c3474c696bb622a861325802abb27/src/Reference/HasOne.php#L243

just because it will exit here : 
https://github.com/atk4/data/blob/0036a94d140c3474c696bb622a861325802abb27/src/Reference/HasOne.php#L227
or here : 
https://github.com/atk4/data/blob/0036a94d140c3474c696bb622a861325802abb27/src/Reference/HasOne.php#L236